File: palindrome.js
//字串倒裝
const isPalindromeStr = (str) => {
  let left = 0;
  let right = str.length - 1;
  let count = 0;
  while (right > left) {
    if (str[left] !== str[right]) {
      console.log(`it takes ${count} times in false`);
      return false;
    } else {
      left++;
      right--;
      count++;
    }
  }
  console.log(`it takes ${count} times in true`);
  return true;
};

console.log(isPalindromeStr("foobar"));
console.log(isPalindromeStr("awesome"));
console.log(isPalindromeStr("awesome"));
console.log(isPalindromeStr("tacocat"));

File: java/src/demo/Palindrome.java
package demo;

public class Palindrome {
    public static void main(String[] args) {
        System.out.println(isPalindromeStr("aabbada"));
    }

    public static boolean isPalindromeStr(String str){
        int left = 0;
        int right = str.length() - 1;
        while(right > left){
            if(str.charAt(left) != str.charAt(right)){
                return false;
            }
            right --;
            left ++;
        }
        return true;
    }
}
